dmithamo/slickiest-blog-api
# # Secure the API by Setting up JWT-Based Auth

#### What does this PR do?
This PR sets up registration and login of user and protects some routes using [jwt token](https://jwt.io/)
#### Description of Task to be completed
- trim nodemon watchlist
- refactor to make common db fns reusable across modules
- refactor logic at `/stories` into a stories controller
- set up a controller for auth
- set up a `users` collection in db
- set up generation of `jwt` token on signup and login
- protect routes using auth logic as middleware

#### How should this be manually tested?

1. Run the following commands in Terminal

```bash
git clone git@github.com:dmithamo/slickiest-blog-api.git
cd slickiest-blog-api
git checkout ft-set-up-auth
yarn
yarn start:dev
```

2. Attempt to send requests to the API. e.g `POST http://localhost:3000//api/v1/stories/`. An error should be thrown, as no Authorization token is provided.

3. Sign up a user (any params will do at this stage) at `POST http://localhost:3000//api/v1/auth/register`. A token should be returned. Copy this and resend the request in `2.` above with this token in the request headers as an Authorization token.

#### What are the relevant Trello Cards?
N/A
#### Screenshots (if appropriate)
1. No token
![image](https://user-images.githubusercontent.com/33679184/60971272-70ed5000-a32c-11e9-8943-73d34608d50b.png)


2. Register a user
![image](https://user-images.githubusercontent.com/33679184/60971342-9e39fe00-a32c-11e9-8c3d-683ebfe26f06.png)


3. With token!
![image](https://user-images.githubusercontent.com/33679184/60971414-c4f83480-a32c-11e9-8add-c34d24c7465d.png)

#### Anything else?
Nope.
#### Checklist
